### ISSUE TITLE
@trezor/request-manager: catch known/expected errors related to TOR and act accordingly

### ISSUE THREAD

**Issue by szymonlesisz** [trezor/trezor-suite#7151]
Follow up for #7133

Workaround is done only for coinjoin communication but it should be done for any outgoing request.
If tor is enabled and `SocksProxyAgent` is used requests may throw errors on which we can react.

1. errors from `nodejs http` module (by e.code === 'ECONNRESET')
examples: "socket hang up" or "socket disconnected before secure TLS connection was established"
When sending request through a keep-alive enabled agent the underlying socket might be reused.
But if server closes connection at unfortunate time client may run into a 'ECONNRESET' error.
each case explanation in ./node_modules/@types/node/*/http.d.ts file

2. errors from `socks` module (by e.type and e.message)
examples: "Socks5 proxy rejected connection" or "Proxy connection timed out"
Basically it means that underlying socket is dead and should be recreated.
error messages to catch listed in ./node_modules/socks/build/common/constants

In both cases currently used circuit should be closed, current agent should be destroyed, new SocksProxyAgent should be created for given identity and request should be repeated automatically

**Comment by karliatto** [comment]
This is not an issue anymore.
